src/api/rest/v1/llm/controller.ts

```diff
@@ -218,6 +232,24 @@ export class LLMController {
             res.end()
         }
     }
+
+    public async agent(req: Request, res: Response) {
+        try {
+            const { context } = await Utils.getNetworkConnectionFromRequest(req)
+            const temperature = req.body.temperature ? parseInt(req.body.temperature.toString()) : 0
+
+            const rag = new Agent()
+            const result = await rag.run(req.body.prompt, context, temperature)
+            return res.json(result)
```

Comment:
`result` is of type `any`, we should strengthen the type and validate the result of `rag.run`  because the consumers of that endpoint will expect a stable result object thyey can rely on